File: functions/src/models/UserLocation.ts
export class UserLocation {
    latitude: number;
    longitude: number;

    constructor(lat?: number, lng?: number) {
        this.latitude = 0;
        this.longitude = 0;
        if (lat != null)
            this.latitude = lat;
        if (lng != null)
            this.longitude = lng;
    }

    static toUrlParameter(location: UserLocation): string {
        return location.latitude.toString() + "%2C" + location.longitude.toString();
    }
}

File: lib/models/UserLocation.dart
import 'package:google_maps_flutter/google_maps_flutter.dart';

class UserLocation {
  late double latitude;
  late double longitude;

  UserLocation({required this.latitude, required this.longitude});

  UserLocation.defaultValues() {
    this.longitude = 1.1;
    this.latitude = 1.1;
  }

  ///LocationData or LatLng
  UserLocation.fromDynamic(dynamic location) {
    this.latitude = location.latitude!;
    this.longitude = location.longitude!;
  }

  UserLocation.fromJson(Map<String, dynamic> json) {
    this.latitude = json['latitude'];
    this.longitude = json['longitude'];
  }

  Map<String, Object?> toJson() =>
      {'latitude': latitude, 'longitude': longitude};

  LatLng toLatLng() {
    return LatLng(latitude, longitude);
  }

  @override
  String toString() => "($latitude,$longitude)";

  String toUrlParameter() {
    return latitude.toString() + '-' + longitude.toString();
  }
}
